### PR TITLE
Fix transactionsRoot return value in case of empty transactions array

### LINKED ISSUE #179: eth_getBlockByNumber expected transactions, none returned (or incorrect transactionsRoot)
### Description

Hi @Nana-EC, I am facing the next obstacle while trying to get `eth_getBlockByNumber` to work. I am consuming the relay json-rpc with `go-ethereum`.

While I try to get the block by number, the library throws a validation error: `server returned empty transaction list but block header indicates transactions`. I have traced the error to this validation step of the go client ~~https://github.com/ethereum/go-ethereum/blob/594e32166269eed4f5cb8270bba99fa234a41606/ethclient/ethclient.go#L134-L136~~
 https://github.com/ethereum/go-ethereum/blob/594e32166269eed4f5cb8270bba99fa234a41606/ethclient/ethclient.go#L137-L139 

When I manually perform a curl to inspect the json-rpc response, I can confirm that `transactions` is an empty array while `transactionsRoot` indicates otherwise.

I expect that a block with no transactions should have a transactions root value as defined here https://github.com/ethereum/go-ethereum/blob/master/core/types/block.go#L35

### Steps to reproduce

1. Run local relay node connected to public testnet mirror node
2. Run go test case as defined in "Additional context"

### Curl to inspect json rpc response:
```sh
curl -X POST http://localhost:7546 \
                                       -H 'Content-Type: application/json' \
                                       -d '{"id":"2","jsonrpc":"2.0","method":"eth_getBlockByNumber","params":[21370223,true]}' --silent | jq
{
  "result": {
    "timestamp": "0x0",
    "difficulty": "0x0",
    "extraData": "0x",
    "gasLimit": "0x0",
    "baseFeePerGas": "0x0",
    "gasUsed": "0x0",
    "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "miner": "0x0000000000000000000000000000000000000000",
    "mixHash": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
    "nonce": "0x0000000000000000",
    "receiptsRoot": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
    "sha3Uncles": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
    "size": "0x0",
    "stateRoot": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
    "totalDifficulty": "0x0",
    "transactions": [],
    "transactionsRoot": "0x7983627d806b9888f818591ebb4dee99ceec9f9e3808f6e4ee0e85ccd9309fad",
    "uncles": [],
    "number": "0x146156f",
    "hash": "0x7983627d806b9888f818591ebb4dee99ceec9f9e3808f6e4ee0e85ccd9309fad",
    "parentHash": "0x60fa091d11691db6f2cf05bd30359384c71e51c32f69fa2ea9ca27f350d5d59c"
  },
  "jsonrpc": "2.0",
  "id": "2"
}
```

### Go test case to reproduce error
```go
package main

import (
	"context"
	"math/big"
	"testing"

	"github.com/ethereum/go-ethereum"
	"github.com/ethereum/go-ethereum/common"
	"github.com/ethereum/go-ethereum/ethclient"
	"github.com/ethereum/go-ethereum/rpc"
	"github.com/stretchr/testify/assert"

	log "github.com/sirupsen/logrus"
)

func createRelayEthClient() *ethclient.Client {
	rpcClient, err := rpc.Dial("http://localhost:7546")
	if err != nil {
		log.Panicf("error connecting to hedera relay node: %s", err.Error())
	}
	return ethclient.NewClient(rpcClient)
}

func TestBlockByNumber(t *testing.T) {
	ethClient := createRelayEthClient()
	defer ethClient.Close()

	blockNumber := big.NewInt(21370223)
	block, err := ethClient.BlockByNumber(context.Background(), blockNumber)
	if err != nil {
		t.Error(err)
		return
	}
	assert.Equal(t, blockNumber, block.Number())
}
```

### Logs from test run
```sh
relay_1  | [2022-06-09 09:04:08.956 +0000] DEBUG (rpc-server/51 on f3e809c6d936): eth_getBlockByNumber
relay_1  | [2022-06-09 09:04:08.956 +0000] TRACE (relay-eth/51 on f3e809c6d936): getBlockByNumber(blockNum=21370223, showDetails=%o)
relay_1  | [2022-06-09 09:04:08.956 +0000] DEBUG (mirror-node/51 on f3e809c6d936): Mirror Request: blocks/21370223
relay_1  | [2022-06-09 09:04:09.529 +0000] DEBUG (mirror-node/51 on f3e809c6d936): Mirror Request: contracts/results?timestamp=gte:1653669740.939629000&timestamp=lte:1653669740.939629000
relay_1  | [2022-06-09 09:04:10.047 +0000] INFO (rpc-server/51 on f3e809c6d936): [POST] eth_getBlockByNumber: 1091 ms Success
```

### Hedera network

testnet

### Version

ghcr.io/hashgraph/hedera-json-rpc-relay:main

### Operating system

Linux

### PR DESCRIPTION
**Pull request by mmrt**
Signed-off-by: Mitchell Martin <mitch@swirldslabs.com>

**Description**:
Return the [proper transactionRoot](https://eth.wiki/json-rpc/API#:~:text=%22transactionsRoot%22%3A%20%220x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421%22%2C) if no transactions exist (instead of blockhash). 

**Related issue(s)**:

Fixes #179

**Result with correct transactionsRoot**:
```
curl -X POST http://localhost:7546 \
                                        -H 'Content-Type: application/json' \
                                        -d '{"id":"2","jsonrpc":"2.0","method":"eth_getBlockByNumber","params":[21370223,true]}' --silent | jq
{
  "result": {
    "timestamp": "0x0",
    "difficulty": "0x0",
    "extraData": "0x",
    "gasLimit": "0x0",
    "baseFeePerGas": "0x0",
    "gasUsed": "0x0",
    "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "miner": "0x0000000000000000000000000000000000000000",
    "mixHash": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
    "nonce": "0x0000000000000000",
    "receiptsRoot": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
    "sha3Uncles": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
    "size": "0x0",
    "stateRoot": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
    "totalDifficulty": "0x0",
    "transactions": [],
    "transactionsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
    "uncles": [],
    "number": "0x146156f",
    "hash": "0x7983627d806b9888f818591ebb4dee99ceec9f9e3808f6e4ee0e85ccd9309fad",
    "parentHash": "0x60fa091d11691db6f2cf05bd30359384c71e51c32f69fa2ea9ca27f350d5d59c"
  },
  "jsonrpc": "2.0",
  "id": "2"
}
```

### CODE REVIEW

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -840,8 +842,8 @@ export class EthImpl implements Eth {
       size: EthImpl.numberTo0x(blockResponse.size | 0),
       stateRoot: EthImpl.emptyArrayHex,
       totalDifficulty: EthImpl.zeroHex,
-      transactions: showDetails ? transactionObjects : transactionHashes,
-      transactionsRoot: blockHash,
+      transactions: transactionArray,
+      transactionsRoot: transactionArray.length == 0 ? EthImpl.ethEmptyTrie : blockHash,
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:846]: This needs coverage in `eth.spec.ts`

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looks good but you're missing tests

File: packages/relay/src/lib/eth.ts (hunk 1 above)
Comment (mmrt) [reply packages/relay/src/lib/eth.ts:846]: coverage now exists

File: packages/relay/tests/lib/eth.spec.ts [hunk 2]
```diff
@@ -371,6 +378,30 @@ describe('Eth calls using MirrorNode', async function () {
     verifyBlockConstants(result);
   });
 
+  it('eth_getBlockByNumber with zero transactions', async function () {
+    // mirror node request mocks
+    mock.onGet(`blocks/${blockNumber}`).reply(200, defaultBlock);
+    mock.onGet(`contracts/results?timestamp=gte:${defaultBlock.timestamp.from}&timestamp=lte:${defaultBlock.timestamp.to}`).reply(200, defaultContractResults);
```
Comment (Nana-EC) [inline packages/relay/tests/lib/eth.spec.ts:384]: A better way would be modify defaultContractResults to return an empty array here.
In this way the follow up `contracts/${contractAddress1}/results` are not made and the mocking is not needed. 
```suggestion
    mock.onGet(`contracts/results?timestamp=gte:${defaultBlock.timestamp.from}&timestamp=lte:${defaultBlock.timestamp.to}`).reply(200, { 'results': []});
```

File: packages/relay/tests/lib/eth.spec.ts [hunk 3]
```diff
@@ -216,6 +216,13 @@ describe('Eth calls using MirrorNode', async function () {
     'v': 1
   };
 
+  const defaultEmptyContractResults = {
```
Comment (Nana-EC) [inline packages/relay/tests/lib/eth.spec.ts:219]: not needed with my suggestion below

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looks good, but I think there's a simpler way on the mocking side.
Should be quick fix and we can get this in

File: packages/relay/tests/lib/eth.spec.ts (hunk 3 above)
Comment (mmrt) [reply packages/relay/tests/lib/eth.spec.ts]: removed

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**